lib/elasticsearch_models/base.rb

```diff
@@ -37,7 +37,7 @@ def create!(*params)
       end
 
       def where(**params)
-        Query::Response.new(client_connection.search(query_params(**params)))
+        Query::Response.new(client_connection.search(query_params(**params)), self)
```

Comment:
One thing I just realized, in `#query_params`, `query_types: type` is added the search params so that the classes can be queried correctly.

e.g. in order to support
```
class Pizza < Food; end
class Burger < Food; end

Food.where # => will return both Pizza and Burger instances
```

If `Burger` is updated to be `Sandwich`, `Food.where` would not return the instances that were created under the `Burger` class.

So in order to support that we'll likely need to change `query_params` to not add `query_params: type` but rather something like `query_params: search_types` with
```
def search_types
  [type]
end
```
So that `Sandwich` could overwrite it so `Food.where` and `Sandwich.where` would correctly return instances that were created when it was named `Burger`
```
def search_types
  ["Burger", *super]
end
```

This should be ok because `Query:Builder` would convert this into an OR.

This change does increase the scope of this ticket, but without it the `model_class_from_name` and `class_factory` changes you made won't actually be used since the queries won't return "old" models (e.g. instances of `Sandwich` that were created when the class was called `Burger`) anyway.

Reply:
>If Burger is updated to be Sandwich, Food.where would not return the instances that were created under the Burger class.
This is not true. The query types of `Burger` that were stored would be `[Burger, Food]`.
So `Food.where` would still return `Burger` documents.

The only edge case we would have to worry about it is the switch from Burger to Sandwich and querying on Sandwich.

Reply:
Ah good catch

Reply:
I was able to reproduce the edge case and I have pushed a fix.